Framework/Algorithms/src/Rebin.cpp

```diff
@@ -250,27 +251,15 @@ void Rebin::exec() {
     PARALLEL_FOR2(inputWS, outputWS)
     for (int hist = 0; hist < histnumber; ++hist) {
       PARALLEL_START_INTERUPT_REGION
-      // get const references to input Workspace arrays (no copying)
-      auto &XValues = inputWS->x(hist).rawData();
-      auto &YValues = inputWS->y(hist).rawData();
-      auto &YErrors = inputWS->e(hist).rawData();
-
-      // get references to output workspace data (no copying)
-      MantidVec &YValues_new = outputWS->dataY(hist);
-      MantidVec &YErrors_new = outputWS->dataE(hist);
-
-      // output data arrays are implicitly filled by function
       try {
-        VectorHelper::rebin(XValues, YValues, YErrors, XValues_new.rawData(),
-                            YValues_new, YErrors_new, dist);
+        outputWS->setHistogram(
+            hist, std::move(HistogramData::rebin(inputWS->histogram(hist),
+                                                 XValues_new)));
       } catch (std::exception &ex) {
         g_log.error() << "Error in rebin function: " << ex.what() << '\n';
         throw;
       }
```

Comment:
Is there a point in catching and rethrowing? I think `Algorithm` will just do the same, so I see little benefit.